File: 2025-05-19/dz1.py
#!/usr/bin/python3

# Задание 1
#with open('text.txt', 'w') as f: f.write('banana mango me the math mango-banana guava-ti-python')
with open('text.txt', 'r') as f: buffer = f.read()
with open('dz_result1.txt', 'w') as f:
    for _ in buffer.split(' '):
        if len(_) > 6: f.writelines(f'{_}\n')

# Задание 2
mass = []
with open('text.txt', 'r') as f:
    for _ in f: mass.append(_.strip())
with open('dz_result2.txt', 'w') as f:
    for _ in mass: f.writelines(f'{_}\n')

# Задание 3
mass.reverse()
with open('dz_result3.txt', 'w') as f:
    for _ in mass: f.writelines(f'{_}\n')

# Задание 4
Flag = True
for _ in range(len(mass)):
    if ',' in mass[_] and Flag: mass.insert(_, '*' * 12); Flag = False
if Flag: mass.append('*' * 12)
mass.reverse()
with open('dz_result4.txt', 'w') as f:
    for _ in mass: f.write(f'{_}\n')

# Задание 5
s = input('Введите символ: ')
i = 0
for _ in mass:
    for l in _.split(' '):
       if l[0] == s: i += 1
print(f'Задание 5: Резульатат: {i}\n')

# Задание 6
with open('dz_result6.txt', 'w') as f:
    for _ in mass:
        if _ != '*' * 12: f.write(f'{_.replace("*", "&")}\n')

# Задание 7
mass = {'Mama mils ramu', 'Uma turman', 'Na pole tanki grohotali', 'Ura!!!', 'Ya ne narkoman'}
with open('dz_result7.txt', 'w') as f:
    for _ in mass: f.write(f'{_}\n')

# Задание 8
with open('dz_result7.txt', 'w') as f:
    for _ in mass: f.write(f'{_}\n')

# Задание 9
with open('text.txt', 'r') as f: buffer = f.read()
print(f'Задание 9: {len(buffer)}')

# Задание 9
with open('text.txt', 'r') as f: buffer = f.read()
print(f'Задание 9: {len(buffer)}')

# Задание 10
mass = []
with open('text.txt', 'r') as f:
    for _ in f: mass.append(_.strip())
print(f'Задание 10: {len(mass)}')

File: 2025-05-19/dz1.go
package main

import (
    "fmt"
    "os"
    "strings"
    "slices"
)

func FileRead(name string) string {
    readFile, _ := os.ReadFile(name)
    return string(readFile)
}

// Функция ввоад числа с проверкой
func Num(txt string) string {
    var p string
    fmt.Printf("%s: ", txt)
    fmt.Scanln(&p)
    return p
}

func Revers(arr []string) []string {
    reversed := []string{}
    for i := 0; i < len(arr); i++ {
        reversed = append(reversed, arr[len(arr)-1-i])
    }
    return reversed
}

func main() {
    // Задание 1
    data := FileRead("text.txt")
    f, _ := os.Create("dz1_result1.txt")
    defer f.Close()
    for _, x := range strings.Split(data, "\n") {
        for _, y := range strings.Split(x, " ") {
            if len(y) > 6 { f.WriteString(y + "\n") }
        }
    }

    // Задание 2
    f, _ = os.Create("dz1_result2.txt")
    defer f.Close()
    for _, x := range strings.Split(data, "\n") { f.WriteString(x + "\n") }

    // Задание 3
    mass := Revers(strings.Split(data, "\n"))
    f, _ = os.Create("dz1_result3.txt")
    defer f.Close()
    for _, x := range mass { f.WriteString(x + "\n") }

    // Задание 4
    Flag := true
    for i :=0; i < len(mass); i++ {
        if strings.Contains(mass[i], ",") && Flag { mass = slices.Insert(mass, i, "************"); Flag = false }
    }
    if Flag { mass = append(mass, "************") }
    mass = Revers(mass)
    f, _ = os.Create("dz1_result4.txt")
    defer f.Close()
    for _, x := range mass { f.WriteString(x + "\n") }

    // Задание 5
    mass = Revers(strings.Split(data, "\n"))
    fmt.Println("Задание: 5")
    s := Num("Введите символ")
    i := 0
    for _, x := range mass {
        for _, y := range strings.Split(x, " ") {
            if strings.HasPrefix(y, s) { i += 1 }
        }
    }
    fmt.Printf("Резульатат: %d\n", i)

    // Задание 6
    f, _ = os.Create("dz1_result6.txt")
    defer f.Close()
    f.WriteString(strings.ReplaceAll(data, "*", "&"))

    // Задание 7
    mass = []string{"Mama mils ramu", "Uma turman", "Na pole tanki grohotali", "Ura!!!", "Ya ne narkoman" }
    f, _ = os.Create("dz1_result7.txt")
    defer f.Close()
    for _, x := range mass { f.WriteString(x + "\n") }

    // Задание 8
    f, _ = os.Create("dz1_result8.txt")
    defer f.Close()
    for _, x := range mass { f.WriteString(x + "\n") }

    // Задание 9
    fmt.Printf("\nЗадание 9\nКоличество символов: %d\n", len(data))

    // Задание 10
    fmt.Printf("\nЗадание 10\nКоличество символов: %d\n", len(strings.Split(data, "\n")))
}